### PR TITLE
Fix ClassLoader issue in test mode with Multipart in multi-module project

### LINKED ISSUE #19037: Multipart Class Defined Outside Resource Implementation Fails with CNFE "Populator"
### Describe the bug

When I define a class meant to be used as a payload in both my REST client and resource I get a CNFE stating that the "populator" could not be located.  If I put the class with the resource it works fine.

Referencing a conversation with Michal on [Google Groups](https://groups.google.com/g/quarkus-dev/c/oq73n_QJzxI/m/zJw0CtJsAAAJ).

**Issue 1 demonstrates the error / Working 2 demonstrates what worked:**

[reproduce-issue1.zip](https://github.com/quarkusio/quarkus/files/6888330/reproduce-issue1.zip)
[reproduce-working2.zip](https://github.com/quarkusio/quarkus/files/6888336/reproduce-working2.zip)

NOTE: Hopefully I did not mix up my reproducers.

### Expected behavior

Class defined in a separate module can be used as the payload from the client and in the resource.

### Actual behavior

Class not found exception - "Populator" when the payload is defined in another module.

ClassNotFoundException: io.platform.api.publish.FilePackage_generated_populator

### How to Reproduce?

Run `mvn clean install` on the issue attached above

### Output of `uname -a` or `ver`

Ubuntu / WSL

### Output of `java -version`

JDK 11 in the POM / 15 as the default.

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.0.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

3.8.1

### Additional information

I forgot to include the Jandex plugin in my reproducer but my modules all are invoking it (and I see the IDX file).  I have also tried adding a `beans.xml` but that did not help.  Anything but the multipart payload seems to work though when separated between an API jar and the service jar.

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #19037

### CODE REVIEW

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/MultipartTransformer.java [hunk 1]
```diff
@@ -79,11 +82,45 @@ public void visitEnd() {
             // ctx param
             injectMethod.visitIntInsn(Opcodes.ALOAD, 1);
 
-            // call the populator
-            injectMethod.visitMethodInsn(Opcodes.INVOKESTATIC, populatorName.replace('.', '/'),
-                    DotNames.POPULATE_METHOD_NAME,
-                    String.format("(%s%s)V", thisDescriptor, INJECTION_CONTEXT_DESCRIPTOR), false);
-
```
Comment (geoand) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/MultipartTransformer.java:86]: I debated whether to keep this around for launch modes that are not test, but I think there isn't too much benefit in doing so...

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/MultipartTransformer.java [hunk 2]
```diff
@@ -79,11 +82,45 @@ public void visitEnd() {
...
+            injectMethod.visitInsn(Opcodes.DUP);
+            injectMethod.visitInsn(Opcodes.ICONST_0);
+            injectMethod.visitVarInsn(Opcodes.ALOAD, 0);
+            injectMethod.visitInsn(Opcodes.AASTORE);
+            injectMethod.visitInsn(Opcodes.DUP);
+            injectMethod.visitInsn(Opcodes.ICONST_1);
+            injectMethod.visitVarInsn(Opcodes.ALOAD, 1);
+            injectMethod.visitInsn(Opcodes.AASTORE);
+            injectMethod.visitMethodInsn(Opcodes.INVOKEVIRTUAL, "java/lang/reflect/Method", "invoke",
+                    "(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;", false);
```
Comment (geoand) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/MultipartTransformer.java:123]: I obviously didn't come up with all this myself... I used the Bytecode Viewer plugin (on a sample class I wrote) in IntelliJ which has the capability to "ASM-ify" bytecode

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]
Let me have a look at what is going on here before we go down this route.

**Comment by stuartwdouglas** [comment]
Superseded by https://github.com/quarkusio/quarkus/pull/19070

**Pull request closed without merging**